File: samples/client/petstore/java/src/main/java/io/swagger/client/Configuration.java
package io.swagger.client;

import java.util.Map;
import java.util.HashMap;

import io.swagger.client.auth.Authentication;
import io.swagger.client.auth.HttpBasicAuth;
import io.swagger.client.auth.ApiKeyAuth;
import io.swagger.client.auth.OAuth;

public class Configuration {
  private static ApiClient defaultApiClient = new ApiClient();

  /**
   * Get the default API client, which would be used when creating API
   * instances without providing an API client.
   */
  public static ApiClient getDefaultApiClient() {
    return defaultApiClient;
   }

  /**
   * Set the default API client, which would be used when creating API
   * instances without providing an API client.
   */
  public static void setDefaultApiClient(ApiClient apiClient) {
    defaultApiClient = apiClient;
  }

  private static final Map<String, Authentication> AUTH;

  static {
    // setup authentications
    AUTH = new HashMap<String, Authentication>();
    
    
    AUTH.put("api_key", new ApiKeyAuth("header", "api_key"));
    
    
    
    
    AUTH.put("petstore_auth", new OAuth());
    
  }

  public static Authentication getAuthentication(String authName) {
    return AUTH.get(authName);
  }

  /** Set username for the first HTTP basic authentication. */
  public static void setUsername(String username) {
    for (Authentication auth : AUTH.values()) {
      if (auth instanceof HttpBasicAuth) {
        ((HttpBasicAuth) auth).setUsername(username);
        return;
      }
    }
    throw new RuntimeException("No HTTP basic authentication configured!");
  }

  /** Set password for the first HTTP basic authentication. */
  public static void setPassword(String password) {
    for (Authentication auth : AUTH.values()) {
      if (auth instanceof HttpBasicAuth) {
        ((HttpBasicAuth) auth).setPassword(password);
        return;
      }
    }
    throw new RuntimeException("No HTTP basic authentication configured!");
  }

  /** Set API key value for the first API key authentication. */
  public static void setApiKey(String apiKey) {
    for (Authentication auth : AUTH.values()) {
      if (auth instanceof ApiKeyAuth) {
        ((ApiKeyAuth) auth).setApiKey(apiKey);
        return;
      }
    }
    throw new RuntimeException("No API key authentication configured!");
  }

  /** Set API key prefix for the first API key authentication. */
  public static void setApiKeyPrefix(String apiKeyPrefix) {
    for (Authentication auth : AUTH.values()) {
      if (auth instanceof ApiKeyAuth) {
        ((ApiKeyAuth) auth).setApiKeyPrefix(apiKeyPrefix);
        return;
      }
    }
    throw new RuntimeException("No API key authentication configured!");
  }
}

File: samples/client/petstore/csharp/src/main/csharp/io/swagger/client/Configuration.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Net;
using System.Text;
using IO.Swagger.Client;

namespace IO.Swagger.Client {
  /// <summary>
  /// Represents a set of configuration settings
  /// </summary>
  public class Configuration{

    /// <summary>
    /// Gets or sets the API client. This is the default API client for making HTTP calls.
    /// </summary>
    /// <value>The API client.</value>
    public static ApiClient apiClient = new ApiClient();

    /// <summary>
    /// Gets or sets the username (HTTP basic authentication)
    /// </summary>
    /// <value>The username.</value>
    public static String username { get; set; }

    /// <summary>
    /// Gets or sets the password (HTTP basic authentication)
    /// </summary>
    /// <value>The password.</value>
    public static String password { get; set; }

    /// <summary>
    /// Gets or sets the API key based on the authentication name 
    /// </summary>
    /// <value>The API key.</value>
    public static Dictionary<String, String> apiKey = new Dictionary<String, String>();

    /// <summary>
    /// Gets or sets the prefix (e.g. Token) of the API key based on the authentication name 
    /// </summary>
    /// <value>The prefix of the API key.</value>
    public static Dictionary<String, String> apiKeyPrefix = new Dictionary<String, String>();


  }
}
